### PR TITLE
F#src 3432 test integrate new driver arms hands bimanual no hands melodic

### PR DESCRIPTION
**Pull request by carebare47**
## Proposed changes

Switches to the new ur10 driver

## Types of changes

What types of changes does your code introduce?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

Put an x in the boxes that apply. You can also fill these out after creating the PR. This is a reminder of what we should look for before merging this code. I have:

- [x] Read and follow the [contributing guidelines](https://github.com/shadow-robot/sr_documentation/blob/master/CONTRIBUTING.md).
- [x] Checked that all tests pass with my changes
- [ ] Added tests (automatic or manual) that prove the fix is effective or that the feature works
- [ ] Added necessary documentation (if appropriate)
- [ ] Added the corresponding license to each file and add the current year of any one that you modified.
- [x] Tested on real hardware (if appropriate)
￼

### CODE REVIEW

File: sr_robot_launch/launch/sr_hardware_control_loop.launch [hunk 1]
```diff
@@ -87,19 +148,22 @@
...
+          - unique_robot_hw
+          - la_sr_ur_robot_hw
+      </rosparam>
+    </group>
+    <rosparam unless="$(arg hand_ctrl)" param="robot_hardware" subst_value="True">[$(arg arm_id)_sr_ur_robot_hw]</rosparam>
     <rosparam command="load" file="$(arg arm_robot_hw)"/>
-    <param name="$(arg arm_prefix)_sr_ur_robot_hw/speed_scale" type="double" value="$(arg arm_speed_scale)"/>
-    <param name="$(arg arm_prefix)_sr_ur_robot_hw/robot_program_path" type="str" value="$(find sr_ur_bringup)/robot_programs/"/>
+    <param name="$(arg arm_id)_sr_ur_robot_hw/speed_scale" type="double" value="$(arg arm_speed_scale)"/>
+    <!--param name="$(arg arm_id)_sr_ur_robot_hw/robot_program_path" type="str" value="$(find sr_ur_bringup)/robot_programs/"/-->
```
Comment (giusebar) [inline sr_robot_launch/launch/sr_hardware_control_loop.launch]: should we remove this commented stuff?

File: sr_robot_launch/launch/sr_hardware_control_loop.launch [hunk 2]
```diff
@@ -114,8 +178,8 @@
     </include>
     <node if="$(arg debug)" name="ur_arm_robot" pkg="ros_control_robot" type="ros_control_robot" args="" output="screen" launch-prefix="gdb -ex run -args"/>
     <node unless="$(arg debug)" name="ur_arm_robot" pkg="ros_control_robot" type="ros_control_robot" args="" output="screen" launch-prefix="ethercat_grant"/>
-    <param name="robot_description" command="xacro --inorder $(arg robot_description) initial_z:=$(arg initial_z)"/>
-    <rosparam command="load" file="$(find sr_ur_robot_hw)/config/$(arg arm_id)_sr_ur_robot_hw.yaml"/>    
+    <param name="robot_description" command="xacro --inorder $(arg robot_description) initial_z:=$(arg initial_z) kinematics_config:=$(arg kinematics_config)"/>
+    <!--rosparam command="load" file="$(find sr_ur_robot_hw)/config/$(arg arm_id)_sr_ur_robot_hw.yaml"/-->
```
Comment (giusebar) [inline sr_robot_launch/launch/sr_hardware_control_loop.launch]: here as well?

**Review by giusebar — CHANGES_REQUESTED** [review]

File: sr_robot_launch/launch/sr_bimanual_hardware_control_loop.launch [hunk 3]
```diff
@@ -53,6 +54,14 @@
   <arg name="arm_position" default="$(eval not arm_trajectory)"/>
   <arg name="right_arm_speed_scale" default="0.5"/>
   <arg name="left_arm_speed_scale" default="0.5"/>
+  <arg name="kinematics_config_left" default="$(find ur_description)/config/ur10_default.yaml"/>
+  <arg name="kinematics_config_right" default="$(find ur_description)/config/ur10_default.yaml"/>
+  <arg name="load_robot_description_command" default="xacro --inorder $(arg robot_description) arm_1_z:=$(arg arm_1_z) arm_2_z:=$(arg arm_2_z) arm_x_separation:=$(arg arm_x_separation) arm_y_separation:=$(arg arm_y_separation) kinematics_config_right:=$(arg kinematics_config_right) kinematics_config_left:=$(arg kinematics_config_left)"/>
+
+  <node if="$(eval arg('hands') == 'none')" pkg="rostopic" type="rostopic" name="hacky_publisher" args="pub /calibrated std_msgs/Bool -- true" output="screen"/>
+
+  <param name="ra_sr_ur_robot_hw/non_blocking_read" type="bool" value="true"/>
```
Comment (giusebar) [inline sr_robot_launch/launch/sr_bimanual_hardware_control_loop.launch]: we should move this non-blocking params in the yaml files

File: sr_robot_launch/launch/sr_hardware_control_loop.launch [hunk 4]
```diff
@@ -51,8 +52,17 @@
     </include>
   </group>
 
+  <!-- So that the driver can check we're loading the right calibration -->
+  <rosparam command="load" file="$(arg kinematics_config)" ns="$(arg arm_prefix)_sr_ur_robot_hw"/>
+
+  <param name="$(arg arm_prefix)_sr_ur_robot_hw/non_blocking_read" type="bool" value="true"/>
```
Comment (giusebar) [inline sr_robot_launch/launch/sr_hardware_control_loop.launch]: here too

**Review by giusebar — DISMISSED** [review]

**Review by beatrizleon — APPROVED** [review]

**Pull request merged**